src/effects/VST3/VST3Effect.cpp

```diff
@@ -350,58 +350,48 @@ bool VST3Effect::SupportsAutomation() const
 bool VST3Effect::SaveSettings(
    const EffectSettings &, CommandParameters & parms) const
 {
-   if(mEditController == nullptr)
-      return false;
-
-   using namespace Steinberg;
-   
-   for(int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
-   {
-      Vst::ParameterInfo parameterInfo { };
-      if(mEditController->getParameterInfo(i, parameterInfo) == kResultOk)
+   return ForEachParameter
+   (
+      [&](const ParameterInfo& pi)
       {
-         if(parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
-         {
-            parms.Write(
-               VST3Utils::MakeAutomationParameterKey(parameterInfo),
-               mEditController->getParamNormalized(parameterInfo.id)
-            );
-         }
-      }
-   }
+         parms.Write(
+            VST3Utils::MakeAutomationParameterKey(pi),
+            mEditController->getParamNormalized(pi.id)
+         );
 
-   return true;
+         return true;
+      }
+   );
 }
 
 bool VST3Effect::LoadSettings(
    const CommandParameters & parms, EffectSettings &settings) const
 {
    using namespace Steinberg;
 
-   if(mComponentHandler == nullptr)
+   if (mComponentHandler == nullptr)
       return false;
-   
-   long index { };
-   wxString key;
-   if(parms.GetFirstEntry(key, index))
-   {
-      do
+
+   ForEachParameter
+   (
+      [&](const ParameterInfo& pi)
       {
-         Steinberg::Vst::ParamID id;
-         Vst::ParamValue value;
-         if(VST3Utils::ParseAutomationParameterKey(key, id) && parms.Read(key, &value))
+         const auto id = pi.id;
```

Comment:
Again:  there should be a first transformation of this function to use `ForEachParameter`.  That means the iteration over `parms` instead becomes a lookup.

`MakeAutomationParameterKey` is used in `SaveSettings` to map `ParameterInfo` to the string key value used in the config file.  This function should call the same, but then to read, not write, that key in the config.  The function `ParseAutomationParameterKey` becomes unused and can be removed.